agent/consul/connect_ca_endpoint.go

```diff
@@ -168,6 +168,16 @@ func (s *ConnectCA) ConfigurationSet(
 		return nil
 	}
 
+	// Have the old provider cross-sign the new intermediate
+	oldProvider, _ := s.srv.getCAProvider()
+	if oldProvider == nil {
+		return fmt.Errorf("internal error: CA provider is nil")
+	}
+
+	if !oldProvider.SupportsCrossSigning() {
+		return fmt.Errorf("error: current CA does not support cross-signing")
```

Comment:
I think the only way we can handle a provider not supporting cross-signing is to just skip that step down below and just proceed without `xcCert` in the intermediates. We'd have to make a note in the docs about the provider not supporting graceful root rotations (since there's no way during the rotation to form a chain from a new leaf to the old root for clients that haven't seen the new root yet).

Reply:
Yeah if you take a look at how we handled this, we made sure config changes when the CA doesn't support this fail unless you explicitly override and accept the temporary connection issues that will result from not cross signing. I think that's a reasonable compromise UX. We should def document (another PR @tradel?) but that at least makes it possible to rotate away from PCA without config just crashing when it attempts to cross sign but only if you acknowledge that doing so may cause connectivity issues.

Reply:
@tradel yeah checked again and we didn't address my last concern on the previous PR - implementing the "override this I know it will cause issues" think we discussed and I described above. Were you planning to do that later in a separate PR? I think it would be best in this PR for completeness of the rotation UX.

Reply:
@banks yes, I was planning to tackle that later. Not because I'm lazy, but because it was getting a bit late to add any more to this PR if we wanted it to land in 1.6.